statesync/syncer_test.go

```diff
@@ -110,7 +110,7 @@ func TestSyncer_SyncAny(t *testing.T) {
 			Hash:   []byte{1},
 		},
 		AppHash: []byte("app_hash_2"),
-	}).Return(&abci.ResponseOfferSnapshot{Result: abci.ResponseOfferSnapshot_REJECT_FORMAT}, nil)
+	}).Return(&abci.ResponseOfferSnapshot{Result: abci.ResponseOfferSnapshot_REJECT_FORMAT}, nil).Maybe()
```

Comment:
I don't trust this change and am a little worried that I'm covering up a problem with this change, but it sort of makes sense to me that this wouldn't get called anymore, as this depended on the ordering in the previous version. I'm not super confident that this method is needed, but I just want to check with someone else in case we need to change the snapshots to make sure this gets called for a reason that's non-obvious to me.

Reply:
I'm not sure if I understand this completely but how come we can't rewrite the tests for the new ranking method

Reply:
I just added a third peer, and added the second snapshot to the third peer so that the peer count doesn't impact the rest of the logic of the test. This is probably good!

If you understand the test better, I def am down to try a different approach.